client/asset/eth/eth.go

```diff
@@ -53,6 +53,7 @@ func registerToken(tokenID uint32, desc string, nets ...dex.Network) {
 	asset.RegisterToken(tokenID, token.Token, &asset.WalletDefinition{
 		Type:        "token",
 		Description: desc,
+		NoAuth:      true,
```

Comment:
Let's work to deprecate this field.

Reply:
Thanks, for the suggestion. I thought about having a separate method `noauth` clones can use but I was hesitant to make many code changes.

Reply:
> I thought about having a separate method noauth clones can use 

Hmm. A method for what?

Reply:
For returning a full node that does not support auth, just like you have in your suggestion `btc.NoAuthClone(...)`.

Reply:
What's the main goal of this undertaking?  I thought it was the useless button on UI, but we have the exact same solution for that, a flag.  If it's the useless Go Lock/Unlock methods, that's not even a problem to solve IMO, unless it turned out to be super trivial. Is there something problematic in core that having the separate Authenticator type solves for the backend, or is this all about the frontend/walletdefinition?

Reply:
> What's the main goal of this undertaking? 

1. Remove the `noauth` field returned in the wallet definition. But seems there's no better way to hide the wallet password input for `zec` for now without explicitly specifying the `noauth` field.
2. Remove unless Lock/Unlock methods. This could seem to be trivial but calling Go methods on wallets that clearly don't implement them didn't match our existing design, hence the `TODO` in `asset/eth/eth.go`.
3. Future wallet implementations that do not require authentications will not have to implement a method they clearly don't need.